doc/source/dev_guide/plugins.rst

```diff
@@ -1,34 +1,207 @@
+``satpy`` so if someone installs it Satpy will automatically be installed.
+The second ``tools.setuptools`` section
+tells the package building (via ``setuptools``) what directory the Python
+code is in. The third section, ``build-system``, says what tool(s) should be
+used for building the package and what extra requirements are needed during
+this build process.
+
+The last section, ``project.entry-points."satpy.composites"`` is the only
+section specific to this package being a Satpy plugin. At the time of writing
+the ``example_composites = "satpy_myplugin"`` portion is not actually used
+by Satpy. Instead Satpy will assume that a package that defines the
+``satpy.composites`` (or any of the other component types) entry point will
+have a ``etc/`` directory in the root of the package structure. Even so,
+for future compatibility, it is best to use the name of the package directory
+on the right-hand side of the ``=``.
+
+.. warning::
+
+    Due to some limitations in setuptools you must also define a ``setup.py``
+    file in addition to ``pyproject.toml`` if you'd like to use "editable"
+    installations (``pip install -e .``). Once
+    `this setuptools issue <https://github.com/pypa/setuptools/issues/2816>`_
+    is resolved this won't be needed. For now this minimal ``setup.py`` will
+    work:
+
+    .. code-block:: python
+
+        from setuptools import setup
+        setup()
+
+**Alternative: setup.py**
+
+If you are more comfortable creating a ``setup.py``-based python package you
+can use ``setup.py`` instead of ``pyproject.toml``. When used for custom
+composites, in a package called ``satpy-myplugin`` it would look something like
+this:
+
+.. code:: python
 
     from setuptools import setup
     import os
 
     setup(
-        name='satpy_cpe',
+        name='satpy-myplugin',
         entry_points={
             'satpy.composites': [
-                'example_composites = satpy_cpe',
+                'example_composites = satpy_myplugin',
             ],
         },
-        package_data={'satpy_cpe': [os.path.join('etc', 'composites/*.yaml')]},
+        package_data={'satpy_myplugin': [os.path.join('etc', 'composites/*.yaml')]},
+        install_requires=["satpy"],
     )
+
+Note the difference between the usage of the package name (``satpy-plugin``)
+which includes a hyphen and the package directory (``satpy_plugin``) which uses
+an underscore. Your package name does not need to have a separator (hyphen) in
+it, but is used here due to the common practice of naming plugins this way.
+See the ``pyproject.toml`` information above for more information on what each
+of these values means.
+
+Licenses
+--------
+
+Disclaimer: We are not lawyers.
+
+Satpy source code is under the GPLv3 license. This license requires any
+derivative works to also be GPLv3. It is our understanding that importing
+a Python module could be considered "linking" that source code to your own
```

Comment:
`could be considered "linking" that source code to your own (and thus being derivative work)...`